pkg/cmd/cli/cmd/rollback.go

```diff
@@ -0,0 +1,85 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kubecmd "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd"
+
+	latestapi "github.com/openshift/origin/pkg/api/latest"
+	client "github.com/openshift/origin/pkg/cmd/client"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+const rollbackLongDesc = `
+Rolls back a deploymentConfig to match a previous deployment. By default, only the pod
+template of the specified deployment will be included in the rollback.
+
+Examples:
+  $ %[1]s %[2]s --from=deployment-2
+  ...
+
+  $ %[1]s %[2]s --from=deployment-3 --include-strategy=true
+  ...
+
+  $ %[1]s %[2]s --from=deployment-1 --output=json
+  ...
+`
+
+func NewCmdRollback(parentName string, name string, f *Factory, out io.Writer) *cobra.Command {
+	rollback := &deployapi.DeploymentConfigRollback{}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s --from=<deployment>", name),
+		Short: "Rolls back a deploymentConfig to match a previous deployment.",
+		Long:  fmt.Sprintf(rollbackLongDesc, parentName, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			from := kubecmd.GetFlagString(cmd, "from")
+			if len(from) == 0 {
+				usageError(cmd, "The --from argument is required.")
+			}
+
+			output := kubecmd.GetFlagString(cmd, "output")
+
+			osClient, _, err := f.Clients(cmd)
+			checkErr(err)
+
+			namespace := getOriginNamespace(cmd)
+
+			newConfig, err := osClient.DeploymentConfigs(namespace).Rollback(rollback)
+			if err != nil {
+				glog.Fatalf("Error generating deploymentConfig for rollback: %v", err)
+			}
+
+			if output == "json" {
+				json, encodeErr := latestapi.Codec.Encode(newConfig)
+				if encodeErr != nil {
+					glog.Fatalf("Couldn't interpret rollback response: %v", err)
+				}
+
+				printer := client.JSONPrinter{}
+				if err := printer.Print(json, out); err != nil {
+					glog.Fatalf("unable to print deploymentConfig JSON: %v [%s]", err, json)
+				}
+
+				return
+			}
+
+			if _, err := osClient.DeploymentConfigs(namespace).Update(newConfig); err != nil {
+				glog.Fatalf("Error performing rollback: %v", err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&rollback.Spec.From.Name, "from", "", "The name of a deployment (replicationController) resource")
+	cmd.Flags().BoolVar(&rollback.Spec.IncludeTemplate, "include-template", true, "Include the previous deployment's pod template in the rollback")
+	cmd.Flags().BoolVar(&rollback.Spec.IncludeTriggers, "include-triggers", false, "Include the previous deployment's triggers in the rollback")
+	cmd.Flags().BoolVar(&rollback.Spec.IncludeStrategy, "include-strategy", false, "Include the previous deployment's strategy in the rollback")
+	cmd.Flags().BoolVar(&rollback.Spec.IncludeReplicationMeta, "include-replication-meta", false, "Include the previous deployment's replicationController metadata in the rollback")
```

Comment:
--include-replica-count-and-selector? Let's not play 20 questions, if you have a name you'd accept, let's use it